plugins/basic/nexus-repository-maven/src/main/java/org/sonatype/nexus/repository/maven/internal/maven2/Maven2MavenPathParser.java

```diff
@@ -106,23 +108,41 @@ else if (str.endsWith("." + HashType.MD5.getExt())) {
           tail = str.substring(nTailPos);
         }
         else {
-          StringBuilder snapshotBuildNumber = new StringBuilder(version);
-          snapshotBuildNumber
-              .append(str.substring(vSnapshotStart + version.length(), vSnapshotStart + version.length() + 8));
+          final StringBuilder snapshotTimestampedVersion = new StringBuilder(version);
```

Comment:
This method's getting a bit complicated, be nice to tease it apart. There's also a lot of magic numbers.